src/main/java/io/vertx/core/impl/DeploymentManager.java

```
@@ -79,13 +99,47 @@ public void deployVerticle(Verticle verticle, DeploymentOptions options,
       throw new IllegalArgumentException("Can't specify isolatedClasses for already created verticle");
     }
     ContextImpl currentContext = vertx.getOrCreateContext();
-    doDeploy("java:" + verticle.getClass().getName(), generateDeploymentID(), options, currentContext, currentContext, completionHandler,
-        getCurrentClassLoader(), verticle);
+    ClassLoader cl = getClassLoader(options, currentContext);
+    currentContext.<Set<Verticle>>executeBlocking(fut -> {
+      int nbInstances = options.getInstances();
+      IdentityHashMap<Class<? extends Verticle>, Set<Verticle>> verticles = new IdentityHashMap<>();
+      for (int i = 0; i < nbInstances; i++) {
+        Verticle verticle = verticleSupplier.get();
```

the case where `get` throws an exception is not handled